crates/http-api-bindings/src/completion/mod.rs

```diff
@@ -59,3 +72,49 @@ pub fn build_completion_prompt(model: &HttpModelConfig) -> (Option<String>, Opti
         (model.prompt_template.clone(), model.chat_template.clone())
     }
 }
+
+fn split_fim_prompt(prompt: &str, support_fim: bool) -> (&str, Option<&str>) {
+    if support_fim {
+        return (prompt, None);
+    }
+
+    let parts = prompt.splitn(2, FIM_TOKEN).collect::<Vec<_>>();
+    (parts[0], parts.get(1).copied())
+}
```

Comment:
```suggestion
fn split_fim_prompt(prompt: &str) -> (&str, Option<&str>) {
    let parts = prompt.splitn(2, FIM_TOKEN).collect::<Vec<_>>();
    (parts[0], parts.get(1).copied())
}
```

Move the condition check to caller.